Diagnostics/HealthChecker/DataCollection/ServerInformation/Get-AllNicInformation.ps1

```diff
@@ -0,0 +1,313 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+. $PSScriptRoot\Get-WmiObjectHandler.ps1
+. $PSScriptRoot\..\..\..\..\Shared\Get-RemoteRegistryValue.ps1
+. $PSScriptRoot\..\..\..\..\Shared\Invoke-CatchActionError.ps1
+. $PSScriptRoot\..\..\..\..\Shared\Invoke-CatchActionErrorLoop.ps1
+Function Get-AllNicInformation {
+    [CmdletBinding()]
+    param(
+        [string]$ComputerName = $env:COMPUTERNAME,
+        [string]$ComputerFQDN,
+        [scriptblock]$CatchActionFunction
+    )
+    begin {
+
+        Function Get-NicPnpCapabilitiesSetting {
+            [CmdletBinding()]
+            param(
+                [string]$NicAdapterComponentId
+            )
+            begin {
+                if ($NicAdapterComponentId -eq [string]::Empty) {
```

Comment:
It would be nice if we could move this validation into the param() declaration. For example, you could ValidateLength to make sure it has at least one character in it. For example:
![image](https://user-images.githubusercontent.com/4518572/127584558-86db429b-d0ee-4ac9-b4f0-a5edd8d2a5b4.png)

Reply:
That is a good point, i will work on looking through all the current Health Checker parameters to see if there is a `throw` based off parameter information that we can replace with a `Validate` in the `param()`

Reply:
Looks like this was the only place under Health Checker that was doing it. 

Fixed in 5c6998e65f876db823c540f08bce57e10f9e2471